include/unifex/scheduler_concepts.hpp

```diff
@@ -341,12 +343,15 @@ namespace _schedule_at {
       (requires receiver<Receiver>)
     friend auto tag_invoke(tag_t<connect>, const type& s, Receiver&& r)
         -> connect_result_t<
-            schedule_at_result_t<std::decay_t<
-                get_scheduler_result_t<const remove_cvref_t<Receiver>&>>&,
+            schedule_at_result_t<
+                get_scheduler_result_t<const remove_cvref_t<Receiver>&>,
                 const TimePoint&>,
             Receiver> {
-      auto scheduler = get_scheduler(std::as_const(r));
-      return connect(schedule_at(scheduler, std::as_const(s.time_point_)), (Receiver&&) r);
+      return connect(
+          schedule_at(
+              get_scheduler(std::as_const(r)),
+              std::as_const(s.time_point_)),
+          (Receiver&&) r);
```

Comment:
This change is dangerous.
If `connect()` customisation happens to take the receiver by-value then the receiver might be in a moved-from state by the time that the `get_scheduler()` CPO is called.
This is why the code was structured the way that it was as two separate statements.